chore: switch to pnpm package manager from yarn

AnujChhikara opened the pull request:
<!-- Date Here in dd mmm yyyy-->
Date: 20 Aug 2025
<!--Developer Name Here-->
Developer Name: @AnujChhikara 

---

## Issue Ticket Number
<!--Issue ticket this PR closes-->
- #2470 
## Description
- This PR migrates the project from Yarn to PNPM as the package manager 

- Successfully deployed to railway https://triumphant-love-production.up.railway.app/
<!--Description of the changes made in this PR-->

### Documentation Updated?

- [x] Yes
- [ ] No

<!--Additional notes about documentation update if applicable-->

### Under Feature Flag

- [ ] Yes
- [x] No

<!--Indicate if changes are under a feature flag-->

### Database Changes

- [ ] Yes
- [x] No

<!--Notes on any database changes-->

### Breaking Changes

- [ ] Yes
- [x] No

<!--Notes on breaking changes or related issue tickets if applicable-->

### Development Tested?

- [x] Yes
- [ ] No

<!--Confirmation of local testing during development-->

## Screenshots
<details>
<summary>Screenshot 1</summary>

<!-- Attach your screenshots here👇-->

https://github.com/user-attachments/assets/f8b03fe3-de90-4f77-b301-79fa21b249f4


</details>

<!--Attach or link to relevant screenshots or visual aids-->

## Test Coverage
<details>
<summary>Screenshot 1</summary>

<!-- Attach your screenshots here👇-->

</details>

<!--Attach Details on test coverage and outcomes-->

## Additional Notes

<!--Any additional notes, considerations or explanations for reviewers-->

<!-- Korbit AI PR Description Start -->
## Description by Korbit AI

### What change is being made?
Switch from using Yarn to pnpm as the package manager, and upgrade several non-breaking package dependencies in the project.

### Why are these changes being made?
Switching to pnpm aims to optimize package management with improved performance and disk space efficiency. Package upgrades ensure compatibility, security compliance, and access to new features, all without introducing breaking changes.

> Is this description stale? Ask me to generate a new description by commenting `/korbit-generate-pr-description`
<!-- Korbit AI PR Description End -->

MayankBansal12 commented:
<img width="542" height="126" alt="Screenshot from 2025-08-25 12-31-10" src="https://github.com/user-attachments/assets/86febbb3-9bf7-4d1e-85e6-a07f0612bc92" />


`@google-cloud/firestore` package wasn't present before, why is it required now?

AnujChhikara commented:
> <img alt="Screenshot from 2025-08-25 12-31-10" width="542" height="126" src="https://private-user-images.githubusercontent.com/117869484/481499745-86febbb3-9bf7-4d1e-85e6-a07f0612bc92.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTYxMDY3OTEsIm5iZiI6MTc1NjEwNjQ5MSwicGF0aCI6Ii8xMTc4Njk0ODQvNDgxNDk5NzQ1LTg2ZmViYmIzLTliZjctNGQxZS04NWU2LWEwN2YwNjEyYmM5Mi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwODI1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDgyNVQwNzIxMzFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yNjI3NDA2ZDk1ZTlhZWM3ZWVhMjg3MmQ1NWYwMWFiYWRjMzAxNGM5ZTg3YTk2NzI5Y2FkZGVmMWNkNGVjZmNlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.mY2RxVT9rcoKeoDPOI7XPSR5hODlaOlsp4760zbFiQc">
> `@google-cloud/firestore` package wasn't present before, how was it working?



> <img alt="Screenshot from 2025-08-25 12-31-10" width="542" height="126" src="https://private-user-images.githubusercontent.com/117869484/481499745-86febbb3-9bf7-4d1e-85e6-a07f0612bc92.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTYxMDY3OTEsIm5iZiI6MTc1NjEwNjQ5MSwicGF0aCI6Ii8xMTc4Njk0ODQvNDgxNDk5NzQ1LTg2ZmViYmIzLTliZjctNGQxZS04NWU2LWEwN2YwNjEyYmM5Mi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwODI1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDgyNVQwNzIxMzFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yNjI3NDA2ZDk1ZTlhZWM3ZWVhMjg3MmQ1NWYwMWFiYWRjMzAxNGM5ZTg3YTk2NzI5Y2FkZGVmMWNkNGVjZmNlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.mY2RxVT9rcoKeoDPOI7XPSR5hODlaOlsp4760zbFiQc">
> `@google-cloud/firestore` package wasn't present before, how was it working?

We need to install @google-cloud/firestore because the latest version of Firestore Admin no longer includes it automatically, and pnpm requires all used packages to be listed directly in your project

Suvidh-kaushik commented on README.md:
````diff
@@ -65,27 +65,27 @@ Head over to [TDD Tests Files List](scripts/tests/tdd-files-list.txt), and add t
+pnpm test
 ```
 
 - Prune dev dependencies
````
we do not need to prune dev dependencies we can directly install  packages using` pnpm install --prod` to install in prod environment

AnujChhikara committed:
chore: migrate from Yarn to pnpm for package management

- Updated documentation in CONTRIBUTING.md and README.md to reflect the use of pnpm instead of Yarn.
- Modified package.json scripts to utilize pnpm commands.
- Added pnpm-lock.yaml for dependency management.
- Removed yarn.lock file to prevent conflicts.
- Updated GitHub Actions workflow to install and test using pnpm.
- Adjusted test scripts to ensure compatibility with pnpm commands.
